src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThicknessConverter.cs

```diff
@@ -169,53 +162,66 @@ static internal string ToString(Thickness th, CultureInfo cultureInfo)
             // Initial capacity [64] is an estimate based on a sum of:
             // 48 = 4x double (twelve digits is generous for the range of values likely)
             //  8 = 4x Unit Type string (approx two characters)
-            //  4 = 4x separator characters
-            StringBuilder sb = new StringBuilder(64);
-
-            sb.Append(LengthConverter.ToString(th.Left, cultureInfo));
-            sb.Append(listSeparator);
-            sb.Append(LengthConverter.ToString(th.Top, cultureInfo));
-            sb.Append(listSeparator);
-            sb.Append(LengthConverter.ToString(th.Right, cultureInfo));
-            sb.Append(listSeparator);
-            sb.Append(LengthConverter.ToString(th.Bottom, cultureInfo));
-            return sb.ToString();
+            //  3 = 3x separator characters
+            //  1 = 1x scratch space for alignment
+
+            DefaultInterpolatedStringHandler handler = new(0, 7, cultureInfo, stackalloc char[64]);
+            FormatDoubleAsString(th.Left, ref handler);
+            handler.AppendFormatted(listSeparator);
+
+            FormatDoubleAsString(th.Top, ref handler);
+            handler.AppendFormatted(listSeparator);
+
+            FormatDoubleAsString(th.Right, ref handler);
+            handler.AppendFormatted(listSeparator);
+
+            FormatDoubleAsString(th.Bottom, ref handler);
+
+            return handler.ToStringAndClear();
         }
 
-        static internal Thickness FromString(string s, CultureInfo cultureInfo)
+        /// <summary> Holds the "Auto" string representation for <see cref="double.NaN"/> conversion. </summary>
+        private static ReadOnlySpan<char> NaNValue => ['A', 'u', 't', 'o'];
```

Comment:
What's the reason for this property ? Usually you just need to convert the string literal to a span.

Reply:
When constructing the final string and checking with `FormatDoubleAsString` for `double.NaN`; using `DefaultInterpolatedStringHandler`, I don't really know the amount of times I'm gonna need to emit the literal string. Any other way it results in boxing of the double or construction of multiple `DefaultInterpolatedStringHandler`, which is both slow and allocates on top.

Reply:
We don't need this, could be a literal, I'll fix that. I just went through different iterations previously and this is a leftover.

Thank you.